packages/expo-in-app-purchases/ios/EXInAppPurchases/EXInAppPurchasesModule.m

```diff
@@ -0,0 +1,346 @@
+{
+  _moduleRegistry = moduleRegistry;
+}
+
+- (NSDictionary *)constantsToExport
+{
+  return @{
+       @"responseCodes": @{
+         @"OK": @OK,
+         @"USER_CANCELED": @USER_CANCELED,
+         @"ERROR": @ERROR,
+         @"SERVICE_DISCONNECTED": @SERVICE_DISCONNECTED
+      },
+      @"purchaseStates": @{
+         @"PURCHASED": [NSNumber numberWithInteger:SKPaymentTransactionStatePurchased],
+         @"PENDING": [NSNumber numberWithInteger:SKPaymentTransactionStatePurchasing],
+         @"FAILED": [NSNumber numberWithInteger:SKPaymentTransactionStateFailed],
+         @"RESTORED": [NSNumber numberWithInteger:SKPaymentTransactionStateRestored],
+         @"DEFERRED": [NSNumber numberWithInteger:SKPaymentTransactionStateDeferred]
+      }
+   };
+}
+
+UM_EXPORT_METHOD_AS(connectToAppStoreAsync,
+                    resolve:(UMPromiseResolveBlock)resolve
+                    reject:(UMPromiseRejectBlock)reject)
+{
+  NSLog(@"Calling ConnectToAppStoreAsync");
+  // Initialize listener and promises dictionary
+  [[SKPaymentQueue defaultQueue] addTransactionObserver:self];
+  promises = [NSMutableDictionary dictionary];
+  queryingItems = NO;
+  [self setPromise:QUERY_HISTORY_KEY resolve:resolve reject:reject];
+
+  // Request history
+  [[SKPaymentQueue defaultQueue] restoreCompletedTransactions];
+}
+
+UM_EXPORT_METHOD_AS(queryPurchasableItemsAsync,
+                    queryPurchasableItemsAsync:(NSArray *)productIDs
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  NSLog(@"Calling queryPurchasableItemsAsync!");
+
+  [self setPromise:QUERY_PURCHASABLE_KEY resolve:resolve reject:reject];
+
+  self->queryingItems = YES;
+  [self requestProducts:productIDs];
+}
+
+UM_EXPORT_METHOD_AS(purchaseItemAsync,
+                    purchaseItemAsync:(NSString *)productIdentifier
+                    resolve:(UMPromiseResolveBlock)resolve
+                    reject:(UMPromiseRejectBlock)reject) {
+  NSLog(@"Calling purchaseItemAsync!");
+
+  if([SKPaymentQueue canMakePayments]){
+    NSLog(@"User can make payments");
+
+    NSArray *productArray = [NSArray arrayWithObjects:productIdentifier,nil];
+    [self setPromise:productIdentifier resolve:resolve reject:reject];
+
+    self->queryingItems = NO;
+    [self requestProducts:productArray];
+  }
+  else{
+    NSLog(@"User cannot make purchases");
+  }
+}
+
+UM_EXPORT_METHOD_AS(queryPurchaseHistoryAsync,
+                    queryPurchaseHistoryAsync:(NSArray *)productIDs
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  NSLog(@"Calling queryPurchasableHistoryAsync!");
+  [self setPromise:QUERY_HISTORY_KEY resolve:resolve reject:reject];
+
+  // Request history
+  [[SKPaymentQueue defaultQueue] restoreCompletedTransactions];
+}
+
+UM_EXPORT_METHOD_AS(disconnectAsync,
+                    resolver:(UMPromiseResolveBlock)resolve
```

Comment:
I'd recommend to define better, more descriptive Obj-C name for this method. Obj-C name also matters, for example when you want to call another exported method from the native side.

It's also pretty close to have a collision here, as you also have this:
```
UM_EXPORT_METHOD_AS(connectToAppStoreAsync,
                     resolve:(UMPromiseResolveBlock)resolve
                     reject:(UMPromiseRejectBlock)reject)
```
which reflects to `resolve:reject:` Obj-C selector whereas the `disconnectAsync` method is `resolver:rejecter:` 😅